drivers/stepper/step_dir/step_dir_stepper_common.h

```diff
@@ -74,11 +77,12 @@ struct step_dir_stepper_common_data {
 	uint64_t microstep_interval_ns;
 	atomic_t actual_position;
 	atomic_t step_count;
+	atomic_t stop_stepper;
 	stepper_event_callback_t callback;
 	void *event_cb_user_data;
 
 	struct k_work_delayable stepper_dwork;
-
+	atomic_t step_completed;
```

Comment:
Nit: This could be further optimised by using a bit array.

```
enum {
	STEP_DIR_STEPPER_STEP_COMPLETED,
	STEP_DIR_STEPPER_STOPPED,
	...
	STEP_DIR_STEPPER_FLAG_NUM_FLAGS,
};


ATOMIC_DEFINE(flags, STEP_DIR_STEPPER_FLAG_NUM_FLAGS);


// Step complete for example
atomic_set_bit(dev->flags, STEP_DIR_STEPPER_STEP_COMPLETED);
```